data/src/main/java/com/linkedin/data/template/WrappingArrayTemplate.java

```diff
@@ -43,15 +43,14 @@ protected WrappingArrayTemplate(DataList list, ArrayDataSchema schema, Class<E>
       throws TemplateOutputCastException
   {
     super(list, schema, elementClass, DataList.class);
```

Comment:
Looks like an existing bug.

Using Object.class will work in theory because no one is using _dataClass right now, but is technically incorrect. I enhanced this code to get the dataclass from the schema.